assertj-core/src/main/java/org/assertj/core/internal/DeepDifference.java

```diff
@@ -263,7 +263,7 @@ private static List<Difference> determineDifferences(Object a, Object b, List<St
       }
 
       // Check List, as element order matters this comparison is faster than using unordered comparison.
-      if (key1 instanceof Collection<?> list) {
+      if (key1 instanceof List<?> list) {
```

Comment:
@timtebeek this is the second one, which originally was:

https://github.com/assertj/assertj/blob/11a36324ea3edb3cb910ee1eb608dc7064e01eb9/assertj-core/src/main/java/org/assertj/core/internal/DeepDifference.java#L267-L268